packages/core-contracts/src/contracts/FleetProxy.sol

```diff
@@ -237,6 +272,56 @@ contract FleetProxy is
                         INTERNAL FUNCTIONS
     //////////////////////////////////////////////////////////////*/
 
+    /**
+     * @notice Gets the source chain ark address from the registry
+     * @param sourceChainId The chain ID where the ark is deployed
+     * @return arkAddress The source chain ark address, or address(0) if not found
+     */
+    function _getSourceChainArk(
+        uint16 sourceChainId
+    ) internal view returns (address arkAddress) {
+        try
+            crossChainRegistry.getArkForProxy(sourceChainId, address(this))
+        returns (address ark) {
+            if (ark != address(0)) {
+                return ark;
+            }
+        } catch {
+            // Registry lookup failed
+        }
+        return address(0);
+    }
+
+    /**
+     * @notice Validates if the source chain is valid for this proxy
+     * @param sourceChainId The chain ID to validate
+     * @return isValid True if the source chain is valid
+     */
+    function _isValidSourceChain(
+        uint16 sourceChainId
+    ) internal view returns (bool isValid) {
```

Comment:
are the silent failures required ? could be reverted here and handeled elswehere on known errors ?